src/Process/3dTilesProcessing.js

```diff
@@ -120,21 +117,13 @@ function computeNodeSSE(camera, node) {
 }
 
 export function init3dTilesLayer(context, layer) {
-    if (layer.initialised) {
-        return;
-    }
-    layer.initialised = true;
-
-    Fetcher.json(layer.url, layer.networkOptions).then((tileset) => {
-        const urlPrefix = layer.url.slice(0, layer.url.lastIndexOf('/') + 1);
-        layer.tileIndex = new $3dTilesIndex(tileset, urlPrefix);
-        layer.asset = tileset.asset;
-        requestNewTile(context.view, context.scheduler, layer, tileset.root, undefined).then(
-            (tile) => {
-                layer.object3d.add(tile);
-                tile.updateMatrixWorld();
-                layer.root = tile;
-            });
+    layer.preprocessedPromise.then((layer) => {
```

Comment:
> What are the implications of a promise being redefined multiple times?

It's not redefined, you just add another callback to its completion. Here it means that the root tiles could be requested several time. I should probably guard the call to requestNewTile (or check if a check is done there).

> More broadly, shouldn't the preprocessedPromise be completed before the layer starts updating?

Imho, not necessarily. We just have to make sure preupdate gives no element back in this case.

> IMHO init3dTilesLayer should be removed completely and its code should move completely in preprocessLayer.

not sure it is possible, but I'll give it a try.

> And maybe MainLoop should be modified to only update layers that have a resolved preprocessedPromise.

if preupdate returns an empty list, it's not necessary. I like the fact preupdate is guaranteed to be called at each loop. WDYT?